### PR TITLE
When using linkerd mesh use `lo` interface for mirroring traffic

### LINKED ISSUE #2452: gRPC traffic is not mirrored when using the Linkerd service mesh
### Bug Description

When debugging a mirrored C# (.Net 8) micro service in Rider, breakpoints for incoming gRPC requests are not hit.

My service listens for HTTP requests on port 8080 and for gRPC requests on port 5000. I see that HTTP handlers are mirrored properly (well, at least the `/healthz` endpoint which is called periodically by the k8s probe). But the gRPC requests are not. Those gRPC requests actually come into the ingress (Contour/Envoy) as grpc-web requests, which are translated to gRPC by Envoy. Then, Envoy sends a meshed request to my service on port 5000. 

Removing the Linkerd proxy from my mirrored pod fixes the issue (my breakpoints are hit)

### Steps to Reproduce

1. Have a Contour ingress, injected with the Linkerd proxy
2. Have a gRPC service in a pod that is also injected by the Linkerd proxy
3. Use mirrord to mirror the service's traffic
4. Make a grpc-web call to the Contour ingress
5. Observe that the mirrored process doesn't receive the request
6. Remove the Linkerd proxy from the gRPC service pod
7. Make a grpc-web call to the Contour ingress
8. Observe that the mirrored process _does_ receive the request

### Backtrace

_No response_

### Relevant Logs

_No response_

### Your operating system and version

macOS 14.4.1 (23E224)

### Local process

A C# (.Net 8) process

### Local process version

_No response_

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by DmitryDodzin**
closes #2452

### CODE REVIEW

File: mirrord/agent/src/sniffer.rs [hunk 1]
```diff
@@ -145,8 +145,8 @@ async fn prepare_sniffer(
     // mesh, otherwise we try to get the appropriate interface.
     let interface = match network_interface.or_else(|| {
         mesh.and_then(|mesh_vendor| match mesh_vendor {
-            MeshVendor::Linkerd | MeshVendor::Kuma => None,
-            MeshVendor::Istio => Some("lo".to_string()),
+            MeshVendor::Linkerd | MeshVendor::Istio => Some("lo".to_string()),
+            MeshVendor::Kuma => None,
```
Comment (aviramha) [inline mirrord/agent/src/sniffer.rs:149]: I think kuma is same, no?

Comment (DmitryDodzin) [reply mirrord/agent/src/sniffer.rs:149]: Yep, I looked at their iptables should also be lo

Comment (DmitryDodzin) [reply mirrord/agent/src/sniffer.rs]: Did update

File: mirrord/agent/src/sniffer.rs [hunk 2]
```diff
@@ -143,12 +143,7 @@ async fn prepare_sniffer(
 ) -> Result<RawCapture, AgentError> {
     // Priority is whatever the user set as an option to mirrord, then we check if we're in an istio
     // mesh, otherwise we try to get the appropriate interface.
-    let interface = match network_interface.or_else(|| {
-        mesh.and_then(|mesh_vendor| match mesh_vendor {
-            MeshVendor::Linkerd | MeshVendor::Kuma => None,
-            MeshVendor::Istio => Some("lo".to_string()),
-        })
-    }) {
+    let interface = match network_interface.or_else(|| mesh.map(|_| "lo".to_string())) {
```
Comment (aviramha) [inline mirrord/agent/src/sniffer.rs:146]: maybe add a comment explanining this

**Review by aviramha — APPROVED** [review]
LGTM, small nitpick.

**Pull request merged**